pkg_resources/__init__.py

```diff
@@ -2023,7 +2114,7 @@ def __init__(self, importer):
 ] = _declare_state('dict', '_distribution_finders', {})
 
 
-def register_finder(importer_type, distribution_finder):
+def register_finder(importer_type: type, distribution_finder: _AdapterType):
```

Comment:
Instead of `_AdapterType`, it would be more "sound" here to have a `callable that, passed a path item and the importer instance, yields ``Distribution`` instances found on that path item` or something that matches the same type as the `_distribution_finders` variable above, would it?

Reply:
See comment https://github.com/pypa/setuptools/pull/4246/files#r1596038268
This would become:
```py
def register_finder(
    importer_type: Type[_T], distribution_finder: _DistFinderType[_T]
) -> None:
```